app/models/review_assignment.rb

```diff
@@ -115,12 +115,11 @@ def reject_by_least_reviewed(contributor_set)
     contributor = contributor_set.min_by {|contributor| contributor.review_mappings.reject {|review_mapping| review_mapping.response.nil? }.count }
     min_reviews = contributor.review_mappings.reject {|review_mapping| review_mapping.response.nil? }.count rescue 0
     contributor_set.reject! {|contributor| contributor.review_mappings.reject {|review_mapping| review_mapping.response.nil? }.count > min_reviews + review_topic_threshold }
-
     contributor_set
   end
 
   def reject_by_max_reviews_per_submission(contributor_set)
-    contributor_set.reject! {|contributor| contributor.review_mappings.reject {|review_mapping| review_mapping.response.nil? }.count >= max_reviews_per_submission }
+    contributor_set.reject! {|contributor| contributor.responses.reject {|response| !response.is_submitted }.count >= max_reviews_per_submission }
```

Comment:
Why did make this change above?

Reply:
Issue #965 says maximum number of reviews per submission is supposed to check the number of reviews that have been submitted, but in the original code it was checking the number of reviews assigned to the reviewers instead. These assigned reviews may not have been submitted. In this new code, am checking the count of review responses of a contributor that are submitted and validating it with max_reviews_per_submission ( obviously it shouldn't cross this value either).